governator-api/src/main/java/com/netflix/governator/event/ApplicationEventDispatcher.java

```diff
@@ -0,0 +1,19 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+
+/**
+ * Interface to publishing {@link ApplicationEvent}s as well as programmatically registering
+ * {@link ApplicationEventListener}s.
+ */
+public interface ApplicationEventDispatcher {
+
+    <T extends ApplicationEvent> void registerListener(Class<T> eventType, ApplicationEventListener<T> eventListener);
+
+    void registerListener(ApplicationEventListener<? extends ApplicationEvent> eventListener);
+
+    void registerListener(Object instance, Method method, Class<? extends ApplicationEvent> acceptedType);
```

Comment:
Should we be exposing the Method based register when we already have the interface ApplicationEventListener.  We can simplify the API always using ApplicationEventListener and create a proxy from method to ApplicationEventListener for @EventListener.